### PR TITLE
layers: Move Pipeline Derivatives to own function

### PR DESCRIPTION
**Pull request by spencer-lunarg**
creates a `ValidatePipelineDerivatives` function to do all pipeline derivatives logic

### CODE REVIEW

File: layers/core_checks/pipeline_validation.cpp [hunk 1]
```diff
@@ -97,9 +96,7 @@ bool CoreChecks::ValidatePipeline(std::vector<std::shared_ptr<PIPELINE_STATE>> c
         const VkPipeline base_handle = pipeline.BasePipeline<VkGraphicsPipelineCreateInfo>();
         const int32_t base_index = pipeline.BasePipelineIndex<VkGraphicsPipelineCreateInfo>();
         if (!((base_handle != VK_NULL_HANDLE) ^ (base_index != -1))) {
-            // TODO: This check is a superset of VUID-VkGraphicsPipelineCreateInfo-flags-00724 and
-            // TODO: VUID-VkGraphicsPipelineCreateInfo-flags-00725
```
Comment (spencer-lunarg) [inline layers/core_checks/pipeline_validation.cpp:101]: https://gitlab.khronos.org/vulkan/vulkan/-/merge_requests/5750 will both reduce this to the first VUID (00724) and then make it so this function can just be used across the various pipeline creations (not just graphics) with same VUID

Comment (artem-lunarg) [reply layers/core_checks/pipeline_validation.cpp:101]: Question about https://gitlab.khronos.org/vulkan/vulkan/-/merge_requests/5750 MR.
The common section replaces the previous definitions, but what will happen with VU IDs? I can't see VUIDs in the new version.

Comment (spencer-lunarg) [reply layers/core_checks/pipeline_validation.cpp:101]: > I can't see VU IDs in the extracted section, how does it work? 

totally a valid question! So the MR doesn't have the new VUID, they will be generated when it gets merged and before being released into the new headers/spec version

from there, these VUID might change, if so, Mike will catch them easily on the header update MR

Comment (artem-lunarg) [reply layers/core_checks/pipeline_validation.cpp:101]: Ah, that's how this spec bakery works. Thanks.

**Review by artem-lunarg — APPROVED** [review]

**Pull request merged**